### DISCUSSION TITLE
BSE data support - Feasibility

### DISCUSSION THREAD

**Discussion by jcopps** [BennyThadikaran/eod2#108] [Ideas]
Hey Benny, 
The repo looks clean and neat. 
I would like to know if there is a plan to include BSE data.

Its been only a hour I have spent. So, am pretty new to this. 
Could one of you point out to the place where it describes all the files stored in the src directory post running init.py.

**Comment by BennyThadikaran** [comment]
Hi Jcopps,

The CSV stock data can be found in `src/eod2_data/daily`. 

I have written a wiki explaining all the details [EOD2 Wiki](https://github.com/BennyThadikaran/eod2/wiki)

`init.py` is used to sync the daily data.

`dget.py` provided stock delivery analysis. The most basic command `py dget.py --sym tcs infy`. This shows todays Delivery qty (DQ), Traded Qty (TQ) and Volume (as multiples) in relation to its 15 day average.

`plot.py` displays charts, a very limited set of indicators and allows drawing trendlines, horizontal lines etc.
To get started run `py plot.py --sym tcs infy`. There are many options available, you may want to read through the wiki.

Ofcourse if you need help or have questions, i am happy to answer them here.

**Comment by BennyThadikaran** [comment]
I used to maintain BSE data (scripts exclusively traded on BSE) in the early days of EOD2.

However, BSE companies eventually get listed on NSE as well. This was creating difficulty maintaining the data in a single folder. 

I dont have immediate plans for this but i am in favor of this idea, provided enough users are interested in it. I would also need feedback on the best way to store and maintain this data.

**Comment by jcopps** [comment]
Great. 
I would like to know how recent actions were handled. 
1. Saturday ad-hoc running of sessions for disaster recovery
2. Market holidays - Temple opening that was not informed prior.

Are these information crucial in the flow.

**Comment by BennyThadikaran** [comment]
In the eod2_data folder you should find a file `meta.json`. The holiday data is stored in this file. The logic is stored in `src/defs/defs.py: checkForHolidays`

Holidays are updated at the start of the year. When syncing, if the date is a holiday it is reupdated (In case holiday was shifted to another day). There is a github issue relating to this. (I cant recall or find it)

1. Saturday ad-hoc running of sessions for disaster recovery
These dates are currently hard coded. Its a temporary fix, since last DR session, I wasnt sure if this will be a repeating feature. 

2. Market holidays - Temple opening that was not informed prior.
This is also currently hard-coded.

Going forward i intend to create a file in the `eod2_data` repo. The script can download the latest file from the repo to check DR sessions. This way users wont have to update their repo and can avoid resulting errors.

**Comment by BennyThadikaran** [comment]
> Its been only a hour I have spent. So, am pretty new to this.
> Could one of you point out to the place where it describes all the files stored in the src directory post running init.py.

I have added an Quick start section in the [Wiki homepage](https://github.com/BennyThadikaran/eod2/wiki). This should help users get familiar before a deep dive into the documentation.
